client/src/classes/PeerProcessor.ts

```diff
@@ -133,6 +141,19 @@ export class PeerProcessor {
   onDataChannelMessage(event: MessageEvent): void {
     const info = JSON.parse(event.data);
     this.addUserInfo(info);
+    if (info.isScreenSharing) {
+      if (!this.screenShareTrigger) {
+        this.screenShareTrigger = true;
+        // If video player doesn't exist, create.
+        this.videoPlayer ??= document.createElement("video");
+        document.getElementById("avatar-video-" + this.peerId)?.appendChild(this.videoPlayer);
```

Comment:
Imo there should be a way to get the user object by id and then call something like `userInfo.byId(this.peerId).addVideo(videoPlayer: HTMLVideoElement)`. Adding stuff to the dom shouldn't be the responsibility of the peerprocessor

Reply:
I totally agree.
I thought about this a little bit but in order to do this, "userInfo" needs to be an actual class instead of just an object and changing this would be way out of the scope for the pr imo.
Also, while I'm at it, I think it would be better to refactor this whole part so that we have a NetworkContext which holds User instances such that it can be used like:
* network.getUsers()
* network.getUserById(id).addVideoPlayer(videoPlayer)
* network.getUsers()[0].name
* network.getUsers()[0].peerProcessor

Reply:
So I'm thinking I'll just do all of that refactoring in a different issue.